### DISCUSSION TITLE
Рабочая конфигурация под дискорд.

### DISCUSSION THREAD

**Discussion by hapugov** [hufrea/byedpi#179] [General]
Привет, у кого то есть рбоачий конфиг для работы разблокировки дискорда?

**Comment by Nikita22007** [comment]
@hapugov 
Я использую следующие параметры
```
 --hosts russia-blacklist.txt --ip 127.0.0.1 --port 1080 --split 1 --disorder 3+s --mod-http=h,d --auto=torst --tlsrec 1+s
```
[russia-blacklist.txt](https://github.com/user-attachments/files/17299407/russia-blacklist.txt)

Но я использую это для своего бота. Я не уверен, можно ли настроить прокси в самом Discord.

UPD:

https://github.com/dovecoteescapee/ByeDPIAndroid/issues/151

```
 --hosts russia-blacklist.txt --ip 127.0.0.1 --port 1080 -s1 -o1 -Ar -o1 -At - f-1 -r1+s -As
```

**Reply by arm64v9a** [reply]
Для прокси в самом дискорде(по крайней мере на linux) есть опция `--proxy-server`, например 
```
discord --proxy-server="socks4://127.0.0.1:1080"
```

**Reply by Drugoy** [reply]
Но с ней не работает голосовое общение.

**Comment by roastme** [comment]
https://p.thenewone.lol/domains-export.txt

**Comment by Horgaring** [comment]
> @hapugov Я использую следующие параметры
> 
> ```
>  --hosts russia-blacklist.txt --ip 127.0.0.1 --port 1080 --split 1 --disorder 3+s --mod-http=h,d --auto=torst --tlsrec 1+s
> ```
> 
> [russia-blacklist.txt](https://github.com/user-attachments/files/17299407/russia-blacklist.txt)
> 
> Но я использую это для своего бота. Я не уверен, можно ли настроить прокси в самом Discord.
> 
> UPD:
> 
> [dovecoteescapee/ByeDPIAndroid#151](https://github.com/dovecoteescapee/ByeDPIAndroid/issues/151)
> 
> ```
>  --hosts russia-blacklist.txt --ip 127.0.0.1 --port 1080 -s1 -o1 -Ar -o1 -At - f-1 -r1+s -As
> ```
войс не работает

**Reply by hapugov** [reply]
> > @hapugov Я использую следующие параметры
> > ```
> >  --hosts russia-blacklist.txt --ip 127.0.0.1 --port 1080 --split 1 --disorder 3+s --mod-http=h,d --auto=torst --tlsrec 1+s
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [russia-blacklist.txt](https://github.com/user-attachments/files/17299407/russia-blacklist.txt)
> > Но я использую это для своего бота. Я не уверен, можно ли настроить прокси в самом Discord.
> > UPD:
> > [dovecoteescapee/ByeDPIAndroid#151](https://github.com/dovecoteescapee/ByeDPIAndroid/issues/151)
> > ```
> >  --hosts russia-blacklist.txt --ip 127.0.0.1 --port 1080 -s1 -o1 -Ar -o1 -At - f-1 -r1+s -As
> > ```
> 
> войс не работает

Ну да, к сожалению. там про запрет что то писали, говорят что с ним работает войс. Могу скинуть ссылку если интересно

**Reply by Nikita22007** [reply]
А если попробовать `-s` заменить на `-s+h` и в конце дописать `-An`?

**Comment by Feanor1397** [comment]
Видел тут в проекте с byedpi для андроид, что у них есть рабочий конфиг для дискорда, с войсом.
https://github.com/dovecoteescapee/ByeDPIAndroid/discussions/173
Решил глянуть что к чему. Настраиваю на роутере, через iptables так и не вышло скормить ему udp.

Собственно после этого решил глянуть как оно устроено на андроид версии, а у них там приложуха поднимает https://github.com/heiher/hev-socks5-tunnel , который умеет udp заворачивать в проксю. Вот только я во всем этом не разбираюсь, вроде делал по инструкции их, но у меня соединение с роутером пропадает после настройки.

Если кто получше шарит и разберется как это чудо на роутере поднять, поделитесь конфигом и как прописывали что с маршрутизацией.

**Reply by YuraU** [reply]
я настроил через iptables, там флаг есть запуска --transparent с ним если запустить, то работает

**Reply by Feanor1397** [reply]
> я настроил через iptables, там флаг есть запуска --transparent с ним если запустить, то работает

Скинь параметры запуска и правила iptables как будет возможность.
У меня выходит так
```
    iptables -t nat -N SOCKS
    iptables -t nat -A SOCKS -d 0.0.0.0/8 -j RETURN
    iptables -t nat -A SOCKS -d 10.0.0.0/8 -j RETURN
    iptables -t nat -A SOCKS -d 100.64.0.0/10 -j RETURN
    iptables -t nat -A SOCKS -d 127.0.0.0/8 -j RETURN
    iptables -t nat -A SOCKS -d 169.254.0.0/16 -j RETURN
    iptables -t nat -A SOCKS -d 172.16.0.0/12 -j RETURN
    iptables -t nat -A SOCKS -d 192.168.0.0/16 -j RETURN
    iptables -t nat -A SOCKS -d 198.18.0.0/15 -j RETURN
    iptables -t nat -A SOCKS -d 224.0.0.0/4 -j RETURN
    iptables -t nat -A SOCKS -d 240.0.0.0/4 -j RETURN
    iptables -t nat -A SOCKS -p tcp -j DNAT --to-destination 127.0.0.1:1080
    iptables -t nat -A SOCKS -p udp -j DNAT --to-destination 127.0.0.1:1080

    iptables -t nat -A PREROUTING -i br0 -p tcp -m tcp -j SOCKS
    iptables -t nat -A PREROUTING -i br0 -p udp -m udp -j SOCKS
```
`ciadpi --transparent --proto=http,tls,udp --disorder=3+s --drop-sack --udp-fake=10 --debug 1`
ИЛИ
`ciadpi --transparent --disorder=3+s --drop-sack --udp-fake=10 --debug 1`

TCP траффик с удовольствием ходит через byedpi, а вот UDP вообще не попадает в него.
Судя по логам дискорда он хочет пойти в 66.22.216.50:50008, но в логах byedpi ни ip ни порт такой не упоминается

**Reply by Feanor1397** [reply]
Окей, походу в transparent режиме он вообще не слушает UDP.

**Reply by YuraU** [reply]
Да, я перепутал, ciadpi в transparent моде у меня слушает только tcp. Посмотрел сейчас. 
Для udp надо делать что-то типа такого, брать udp2raw или redsocks.
redsocks это transparent proxy, который сможет слушать udp и переадресовать уже на ciadpi
А udp2raw, если не ошибаюсь, завернет udp трафик как tcp и его можно опять кинуть на ciadpi

**Comment by DMS-Chiii** [comment]
Какой конфиг сейчас работает?

**Reply by hapugov** [reply]
> Какой конфиг сейчас работает?

Я ставил запрет на линукс, чтоб работали боты. Надо тестировать чтоб понять какой конфиг подойдет. 
А на винду гуглите ytdisbystro, у меня на виндеработает прекрасно, текст и потоковое вещаение летает.

**Comment by frogfile** [comment]
Нашли рабочую конфигурацию для войса?
